clinica/iotools/bids_utils.py

```diff
@@ -523,168 +534,247 @@ def write_modality_agnostic_files(
     _write_bidsignore(bids_dir)
 
 
-def write_sessions_tsv(bids_dir, sessions_dict):
+def write_sessions_tsv(bids_dir: str, sessions_dict: dict) -> None:
     """Create <participant_id>_sessions.tsv files.
 
-    Write the content of the function create scans dict in several TSV files
+    Basically writes the content of the function
+    `clinica.iotools.bids_utils.create_sessions_dict` in several TSV files
     following the BIDS specification.
 
-    Args:
-        bids_dir: path to the bids directory
-        sessions_dict: output of the function create_sessions_dict
+    Parameters
+    ----------
+    bids_dir : str
+        Path to the BIDS directory.
+
+    sessions_dict : dict
+        Dictionary containing sessions metadata.
+
+        .. note::
+            This is the output of the function
+            `clinica.iotools.bids_utils.create_sessions_dict`.
+
+    See also
+    --------
+    create_sessions_dict
+    write_scans_tsv
     """
-    import os
-    from glob import glob
-    from os import path
+    from pathlib import Path
 
     import pandas as pd
 
-    bids_paths = glob(path.join(bids_dir, "sub-*"))
-
-    for sp in bids_paths:
-        bids_id = sp.split(os.sep)[-1]
+    bids_dir = Path(bids_dir)
 
-        if bids_id in sessions_dict:
-            session_df = pd.DataFrame.from_dict(sessions_dict[bids_id], orient="index")
+    for sp in bids_dir.glob("sub-*"):
```

Comment:
Is `sp` short for `subject_path`? If so, better use the latter to help cognition.